src/server/JasmineGraphInstanceService.cpp

```diff
@@ -3584,3 +3650,309 @@ map<long, map<long, unordered_set<long>>> calculateEgoNet(string graphID, string
+
+    for (vector<string>::iterator workerIt = workerSockets.begin(); workerIt != workerSockets.end(); ++workerIt) {
+        std::vector<string> workerSocketPair;
+        stringstream wl(*workerIt);
+        string intermediate;
+        while (getline(wl, intermediate, ':')) {
+            workerSocketPair.push_back(intermediate);
+        }
+
+        Utils utils;
+        std::string aggregatorFilePath = utils.getJasmineGraphProperty("org.jasminegraph.server.instance.datafolder");
+        std::string centralGraphIdentifier = graphID + +"_centralstore_" + workerSocketPair[2];
+
+        std::string centralStoreFile = aggregatorFilePath + "/" + centralGraphIdentifier;
+        instance_logger.log("###INSTANCE### centralstore " + centralStoreFile, "info");
+
+        struct stat s;
+        if (stat(centralStoreFile.c_str(), &s) == 0) {
+            if (s.st_mode & S_IFREG) {
+                JasmineGraphHashMapCentralStore centralStore = JasmineGraphInstanceService::loadCentralStore(
+                        centralStoreFile);
+                map<long, unordered_set<long>> centralGraphMap = centralStore.getUnderlyingHashMap();
+
+                for (map<long, unordered_set<long>>::iterator centralGraphMapIterator = centralGraphMap.begin();
+                     centralGraphMapIterator != centralGraphMap.end(); ++centralGraphMapIterator) {
+                    long startVid = centralGraphMapIterator->first;
+                    unordered_set<long> endVidSet = centralGraphMapIterator->second;
+
+                    for (auto itr = endVidSet.begin(); itr != endVidSet.end(); ++itr) {
+                        map<long, long>::iterator fromDegreeDistIterator = fromDegreeDist.find(startVid);
+                        if (fromDegreeDistIterator != fromDegreeDist.end()) {
+                            long degree = fromDegreeDistIterator->second;
+                            fromDegreeDistIterator->second = degree + 1;
+                        } else {
+                            fromDegreeDist.insert(std::make_pair(startVid, 1));
+                        }
+
+                        if (graphVertexMap.find(*itr) != graphVertexMap.end()) {
+                            map<long, unordered_set<long>>::iterator toIDiterator = worldToLocalVertexMap.find(*itr);
+                            if (toIDiterator != worldToLocalVertexMap.end()) {
+                                unordered_set<long> fromIDs = toIDiterator->second;
+                                fromIDs.insert(startVid);
+                                toIDiterator->second = fromIDs;
+                            } else {
+                                unordered_set<long> fromIDs;
+                                fromIDs.insert(startVid);
+                                worldToLocalVertexMap.insert(std::make_pair(*itr, fromIDs));
+                            }
+                        }
+
+                    }
+                }
+            }
+        }
+    }
+
+    double alpha = 0.85;
```

Comment:
Can this `alpha` value be fed as an input parameter via the frontend command sequence? We can give the default value as 0.85 in the command sequence and ask the user to use that one or specify different value. We should do some validation as well before we continue using the value proposed by user as the alpha.

Reply:
changed

Reply:
Ok